File: TextosJapones/src/main/java/org/jisho/textosJapones/controller/MenuConfiguracaoController.java
package org.jisho.textosJapones.controller;

import com.jfoenix.controls.JFXButton;
import com.jfoenix.controls.JFXPasswordField;
import com.jfoenix.controls.JFXTextField;
import javafx.fxml.FXML;
import javafx.fxml.Initializable;
import javafx.stage.DirectoryChooser;
import org.jisho.textosJapones.database.mysql.ConexaoMysql;
import org.jisho.textosJapones.util.constraints.Validadores;

import java.io.File;
import java.net.URL;
import java.util.ResourceBundle;

public class MenuConfiguracaoController implements Initializable {

	@FXML
	public JFXTextField txtUsuario;

	@FXML
	public JFXPasswordField pswSenha;

	@FXML
	public JFXTextField txtServer;

	@FXML
	public JFXTextField txtPorta;

	@FXML
	public JFXTextField txtDataBase;

	@FXML
	public JFXTextField txtCaminhoMysql;

	@FXML
	public JFXTextField txtCaminhoWinrar;

	@FXML
	public JFXButton btnCaminhoMysql;
	
	@FXML
	public JFXButton btnCaminhoWinrar;

	private MenuPrincipalController controller;

	@FXML
	private void onBtnCarregarCaminhoMysql() {
		controller.getPopPup().setDetached(true);
		String caminho = selecionaPasta("Selecione a pasta do mysql", txtCaminhoMysql.getText());
		txtCaminhoMysql.setText(caminho);
	}
	
	@FXML
	private void onBtnCarregarCaminhoWinrar() {
		controller.getPopPup().setDetached(true);
		txtCaminhoWinrar.setText(selecionaPasta("Selecione a pasta do winrar", txtCaminhoWinrar.getText()));
	}

	public void salvar() {
		ConexaoMysql.setDadosConexao(txtServer.getText(), txtPorta.getText(), txtUsuario.getText(), pswSenha.getText(), 
				txtCaminhoMysql.getText(), txtCaminhoWinrar.getText(), txtDataBase.getText());
		controller.verificaConexao();
	}

	public void carregar() {
		ConexaoMysql.getDadosConexao();
		txtServer.setText(ConexaoMysql.getServer());
		txtPorta.setText(ConexaoMysql.getPort());
		txtDataBase.setText(ConexaoMysql.getDataBase());
		txtUsuario.setText(ConexaoMysql.getUser());
		pswSenha.setText(ConexaoMysql.getPassword());
		txtCaminhoMysql.setText(ConexaoMysql.getCaminhoMysql());
		txtCaminhoWinrar.setText(ConexaoMysql.getCaminhoWinrar());
	}

	private String selecionaPasta(String titulo, String pasta) {
		DirectoryChooser fileChooser = new DirectoryChooser();
		fileChooser.setTitle(titulo);

		if (pasta != null && !pasta.isEmpty())
			fileChooser.setInitialDirectory(new File(pasta));
		File caminho = fileChooser.showDialog(null);

		if (caminho == null)
			return "";
		else
			return caminho.getAbsolutePath();
	}
	
	public void setControllerPai(MenuPrincipalController cnt) {
		this.controller = cnt;
	}

	public static URL getFxmlLocate() {
		return MenuConfiguracaoController.class.getResource("/view/MenuConfiguracao.fxml");
	}

	@Override
	public void initialize(URL location, ResourceBundle resources) {
		Validadores.setTextFieldNotEmpty(txtServer);
		Validadores.setTextFieldNotEmpty(txtPorta);
		Validadores.setTextFieldNotEmpty(txtDataBase);
		Validadores.setTextFieldNotEmpty(txtUsuario);
		Validadores.setTextFieldNotEmpty(pswSenha);
	}
}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/controller/MenuConfiguracaoController.kt
package br.com.fenix.processatexto.controller

import br.com.fenix.processatexto.util.configuration.Configuracao
import br.com.fenix.processatexto.util.constraints.Validadores
import com.jfoenix.controls.JFXButton
import com.jfoenix.controls.JFXPasswordField
import com.jfoenix.controls.JFXTextField
import javafx.fxml.FXML
import javafx.fxml.Initializable
import javafx.stage.DirectoryChooser
import java.io.File
import java.net.URL
import java.util.*


class MenuConfiguracaoController : Initializable {

    @FXML
    lateinit var txtUsuario: JFXTextField

    @FXML
    lateinit var pswSenha: JFXPasswordField

    @FXML
    lateinit var txtServer: JFXTextField

    @FXML
    lateinit var txtPorta: JFXTextField

    @FXML
    lateinit var txtDataBase: JFXTextField

    @FXML
    lateinit var txtCaminhoMysql: JFXTextField

    @FXML
    lateinit var txtCaminhoWinrar: JFXTextField

    @FXML
    lateinit var txtCaminhoCommicTagger: JFXTextField

    @FXML
    lateinit var btnCaminhoMysql: JFXButton

    @FXML
    lateinit var btnCaminhoWinrar: JFXButton

    @FXML
    lateinit var btnCaminhoCommicTagger: JFXButton

    private lateinit var controller: MenuPrincipalController

    @FXML
    private fun onBtnCarregarCaminhoMysql() {
        controller.popPup!!.isDetached = true
        val caminho = selecionaPasta("Selecione a pasta do mysql", txtCaminhoMysql.text)
        txtCaminhoMysql.text = caminho
    }

    @FXML
    private fun onBtnCarregarCaminhoWinrar() {
        controller.popPup!!.isDetached = true
        txtCaminhoWinrar.text = selecionaPasta("Selecione a pasta do winrar", txtCaminhoWinrar.text)
    }

    @FXML
    private fun onBtnCarregarCaminhoCommicTagger() {
        controller.popPup!!.isDetached = true
        txtCaminhoCommicTagger.text = selecionaPasta("Selecione a pasta do commictagger", txtCaminhoCommicTagger.text)
    }

    fun salvar() {
        Configuracao.server = txtServer.text
        Configuracao.port = txtPorta.text
        Configuracao.database = txtDataBase.text
        Configuracao.user = txtUsuario.text
        Configuracao.password = pswSenha.text
        Configuracao.caminhoMysql = txtCaminhoMysql.text
        Configuracao.caminhoWinrar = txtCaminhoWinrar.text
        Configuracao.caminhoCommicTagger = txtCaminhoCommicTagger.text
        Configuracao.saveProperties()
    }

    fun carregar() {
        txtServer.text = Configuracao.server
        txtPorta.text = Configuracao.port
        txtDataBase.text = Configuracao.database
        txtUsuario.text = Configuracao.user
        pswSenha.text = Configuracao.password
        txtCaminhoMysql.text = Configuracao.caminhoMysql
        txtCaminhoWinrar.text = Configuracao.caminhoWinrar
        txtCaminhoCommicTagger.text = Configuracao.caminhoCommicTagger
    }

    private fun selecionaPasta(titulo: String, pasta: String?): String {
        val fileChooser = DirectoryChooser()
        fileChooser.title = titulo

        if (pasta != null && pasta.isNotEmpty())
            fileChooser.initialDirectory = File(pasta)

        val caminho: File = fileChooser.showDialog(null)

        return if (caminho == null) "" else caminho.absolutePath
    }

    fun setControllerPai(cnt: MenuPrincipalController) {
        controller = cnt
    }

    @Override
    override fun initialize(location: URL?, resources: ResourceBundle?) {
        Validadores.setTextFieldNotEmpty(txtServer)
        Validadores.setTextFieldNotEmpty(txtPorta)
        Validadores.setTextFieldNotEmpty(txtDataBase)
        Validadores.setTextFieldNotEmpty(txtUsuario)
        Validadores.setTextFieldNotEmpty(pswSenha)
    }

    companion object {
        val fxmlLocate: URL get() = MenuConfiguracaoController::class.java.getResource("/view/MenuConfiguracao.fxml") as URL
    }
}